File: lint-rules-android-lint/src/main/java/com/vanniktech/lintrules/android/IssueRegistry.java
package com.vanniktech.lintrules.android;

import com.android.tools.lint.detector.api.Issue;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import static com.vanniktech.lintrules.android.AnnotationOrderDetectorKt.ISSUE_WRONG_ANNOTATION_ORDER;
import static com.vanniktech.lintrules.android.ConstraintLayoutToolsEditorAttributeDetectorKt.ISSUE_CONSTRAINT_LAYOUT_TOOLS_EDITOR_ATTRIBUTE_DETECTOR;
import static com.vanniktech.lintrules.android.DefaultLayoutAttributeDetector.ISSUE_DEFAULT_LAYOUT_ATTRIBUTE;
import static com.vanniktech.lintrules.android.InvalidSingleLineCommentDetector.ISSUE_INVALID_SINGLE_LINE_COMMENT;
import static com.vanniktech.lintrules.android.InvalidStringDetector.ISSUE_INVALID_STRING;
import static com.vanniktech.lintrules.android.RawDimenDetectorKt.ISSUE_RAW_DIMEN;
import static com.vanniktech.lintrules.android.SuperfluousNameSpaceDetectorKt.ISSUE_SUPERFLUOUS_NAME_SPACE;
import static com.vanniktech.lintrules.android.UnsupportedLayoutAttributeDetectorKt.ISSUE_UNSUPPORTED_LAYOUT_ATTRIBUTE;
import static com.vanniktech.lintrules.android.WrongLayoutNameDetectorKt.ISSUE_WRONG_LAYOUT_NAME;
import static com.vanniktech.lintrules.android.MatchingMenuIdDetector.ISSUE_MATCHING_MENU_ID;
import static com.vanniktech.lintrules.android.MatchingViewIdDetector.ISSUE_MATCHING_VIEW_ID;
import static com.vanniktech.lintrules.android.MissingXmlHeaderDetectorKt.ISSUE_MISSING_XML_HEADER;
import static com.vanniktech.lintrules.android.RawColorDetector.ISSUE_RAW_COLOR;
import static com.vanniktech.lintrules.android.ShouldUseStaticImportDetector.ISSUE_SHOULD_USE_STATIC_IMPORT;
import static com.vanniktech.lintrules.android.SuperfluousMarginDeclarationDetector.SUPERFLUOUS_MARGIN_DECLARATION;
import static com.vanniktech.lintrules.android.SuperfluousPaddingDeclarationDetector.SUPERFLUOUS_PADDING_DECLARATION;
import static com.vanniktech.lintrules.android.WrongConstraintLayoutUsageDetector.ISSUE_WRONG_CONSTRAINT_LAYOUT_USAGE;
import static com.vanniktech.lintrules.android.WrongMenuIdFormatDetector.ISSUE_WRONG_MENU_ID_FORMAT;
import static com.vanniktech.lintrules.android.WrongTestMethodNameDetectorKt.ISSUE_WRONG_TEST_METHOD_NAME;
import static com.vanniktech.lintrules.android.WrongViewIdFormatDetector.ISSUE_WRONG_VIEW_ID_FORMAT;
import static com.vanniktech.lintrules.android.XmlSpacingDetectorKt.ISSUE_XML_SPACING;

public final class IssueRegistry extends com.android.tools.lint.client.api.IssueRegistry {
  @Override public List<Issue> getIssues() {
    final List<Issue> issues = new ArrayList<>();
    Collections.addAll(issues, AndroidDetector.getIssues());
    issues.add(ISSUE_WRONG_VIEW_ID_FORMAT);
    issues.add(ISSUE_WRONG_MENU_ID_FORMAT);
    issues.add(ISSUE_RAW_DIMEN);
    issues.add(ISSUE_RAW_COLOR);
    issues.add(SUPERFLUOUS_MARGIN_DECLARATION);
    issues.add(SUPERFLUOUS_PADDING_DECLARATION);
    issues.add(ISSUE_SHOULD_USE_STATIC_IMPORT);
    issues.add(ISSUE_MATCHING_VIEW_ID);
    issues.add(ISSUE_MATCHING_MENU_ID);
    issues.add(ISSUE_INVALID_SINGLE_LINE_COMMENT);
    issues.add(ISSUE_INVALID_STRING);
    issues.add(ISSUE_DEFAULT_LAYOUT_ATTRIBUTE);
    issues.add(ISSUE_WRONG_CONSTRAINT_LAYOUT_USAGE);
    issues.add(ISSUE_MISSING_XML_HEADER);
    issues.add(ISSUE_WRONG_TEST_METHOD_NAME);
    issues.add(ISSUE_WRONG_LAYOUT_NAME);
    issues.add(ISSUE_WRONG_ANNOTATION_ORDER);
    issues.add(ISSUE_CONSTRAINT_LAYOUT_TOOLS_EDITOR_ATTRIBUTE_DETECTOR);
    issues.add(ISSUE_SUPERFLUOUS_NAME_SPACE);
    issues.add(ISSUE_XML_SPACING);
    issues.add(ISSUE_UNSUPPORTED_LAYOUT_ATTRIBUTE);
    return issues;
  }
}

File: lint-rules-android-lint/src/main/java/com/vanniktech/lintrules/android/IssueRegistry.kt
package com.vanniktech.lintrules.android

import com.android.tools.lint.detector.api.CURRENT_API
import com.vanniktech.lintrules.android.MatchingMenuIdDetector.ISSUE_MATCHING_MENU_ID
import com.vanniktech.lintrules.android.MatchingViewIdDetector.ISSUE_MATCHING_VIEW_ID
import com.vanniktech.lintrules.android.SuperfluousMarginDeclarationDetector.ISSUE_SUPERFLUOUS_MARGIN_DECLARATION
import com.vanniktech.lintrules.android.SuperfluousPaddingDeclarationDetector.ISSUE_SUPERFLUOUS_PADDING_DECLARATION
import com.vanniktech.lintrules.android.WrongConstraintLayoutUsageDetector.ISSUE_WRONG_CONSTRAINT_LAYOUT_USAGE

internal const val PRIORITY = 10 // Does not matter anyways within Lint.

class IssueRegistry : com.android.tools.lint.client.api.IssueRegistry() {
  override val api = CURRENT_API

  override val issues get() =
      AndroidDetector.getIssues()
      .toList()
      .plus(listOf(
          ISSUE_INVALID_IMPORT,
          ISSUE_WRONG_VIEW_ID_FORMAT,
          ISSUE_WRONG_MENU_ID_FORMAT,
          ISSUE_RAW_DIMEN,
          ISSUE_RAW_COLOR,
          ISSUE_SUPERFLUOUS_MARGIN_DECLARATION,
          ISSUE_SUPERFLUOUS_PADDING_DECLARATION,
          ISSUE_SHOULD_USE_STATIC_IMPORT,
          ISSUE_MATCHING_VIEW_ID,
          ISSUE_MATCHING_MENU_ID,
          ISSUE_INVALID_SINGLE_LINE_COMMENT,
          ISSUE_INVALID_STRING,
          ISSUE_DEFAULT_LAYOUT_ATTRIBUTE,
          ISSUE_TODO,
          ISSUE_WRONG_CONSTRAINT_LAYOUT_USAGE,
          ISSUE_MISSING_XML_HEADER,
          ISSUE_WRONG_TEST_METHOD_NAME,
          ISSUE_WRONG_LAYOUT_NAME,
          ISSUE_WRONG_ANNOTATION_ORDER,
          ISSUE_CONSTRAINT_LAYOUT_TOOLS_EDITOR_ATTRIBUTE_DETECTOR,
          ISSUE_SUPERFLUOUS_NAME_SPACE,
          ISSUE_XML_SPACING,
          ISSUE_NAMING_PATTERN,
          ISSUE_UNUSED_MERGE_ATTRIBUTES,
          ISSUE_LAYOUT_FILE_NAME_MATCHES_CLASS,
          ISSUE_UNSUPPORTED_LAYOUT_ATTRIBUTE))
}
